packages/core/src/shared/sam/sync.ts

```diff
@@ -354,16 +354,24 @@ export class SyncWizard extends Wizard<SyncParams> {
         })
 
         this.form.ecrRepoUri.bindPrompter(({ region }) => createEcrPrompter(new DefaultEcrClient(region!)), {
-            showWhen: ({ template }) => !!template && hasImageBasedResources(template.data),
+            showWhen: ({ template, paramsSource }) =>
+                !!template &&
+                hasImageBasedResources(template.data) &&
+                (paramsSource === ParamsSource.Flags || paramsSource === ParamsSource.SpecifyAndSave),
         })
 
         // todo wrap with localize
-        this.form.syncFlags.bindPrompter(() =>
-            createMultiPick(syncFlagItems, {
-                title: 'Specify parameters for sync',
-                placeholder: 'Press enter to proceed with highlighted option',
-                buttons: createCommonButtons(samSyncParamUrl),
-            })
+        this.form.syncFlags.bindPrompter(
```

Comment:
What would be syncFlags if we skipped here? Will it be empty?

Reply:
I believe that what we discussed last time. 
When the customer use samconfig file, we use all the flag values from the file; this mean, we would use `--config-file` as mention [here](https://github.com/aws/aws-toolkit-vscode/blob/5554065a8c0614a499e259456dafbe91340b15a7/packages/core/src/shared/sam/sync.ts#L424-L460). The based on the value, the [boundArgs](https://github.com/aws/aws-toolkit-vscode/blob/5554065a8c0614a499e259456dafbe91340b15a7/packages/core/src/shared/sam/sync.ts#L544) is loaded here. We do not need the sync flags for merging with `boundArgs` [here]([[fix(lambda): skip ecrRepoUri and syncFlags prompters when using flag from samconfig file  by vicheey · Pull Request #5913 · aws/aws-toolkit-vscode>>https://github.com/aws/aws-toolkit-vscode/pull/5913]]) because again, [--config-file](https://github.com/aws/aws-toolkit-vscode/blob/5554065a8c0614a499e259456dafbe91340b15a7/packages/core/src/shared/sam/sync.ts#L452) is used. 

Please let me know if there is issue.

Reply:
What I mean is, are the sync flags actually empty if they are skipped. Could you help to confirm?

Reply:
Yes, confirm that it is an empty array